bootstore/src/schemes/v0/peer.rs

```diff
@@ -0,0 +1,810 @@
+            }
+            NodeApiRequest::InitLearner { responder } => {
+                if self.init_responder.is_some() {
+                    let _ = responder
+                        .send(Err(NodeRequestError::RequestAlreadyPending));
+                    return;
+                }
+                if let Err(err) = self.fsm.init_learner(Instant::now().into()) {
+                    let _ = responder.send(Err(err.into()));
+                } else {
+                    self.init_responder = Some(responder);
+                    self.deliver_envelopes().await;
+                }
+            }
+            NodeApiRequest::LoadRackSecret { responder } => {
+                if self.rack_secret_responder.is_some() {
+                    let _ = responder
+                        .send(Err(NodeRequestError::RequestAlreadyPending));
+                    return;
+                }
+                if let Err(err) =
+                    self.fsm.load_rack_secret(Instant::now().into())
+                {
+                    let _ = responder.send(Err(err.into()));
+                } else {
+                    self.rack_secret_responder = Some(responder);
+                    self.deliver_envelopes().await;
+                }
+            }
+            NodeApiRequest::PeerAddresses(peers) => {
+                info!(self.log, "Updated Peer Addresses: {peers:?}");
+                self.manage_connections(peers).await;
+            }
+            NodeApiRequest::GetStatus { responder } => {
+                let status = Status {
+                    fsm_state: self.fsm.state().name(),
+                    peers: self.peers.clone(),
+                    connections: self
+                        .established_connections
+                        .iter()
+                        .map(|(id, handle)| (id.clone(), handle.addr))
+                        .collect(),
+                    accepted_connections: self
+                        .accepted_connections
+                        .keys()
+                        .cloned()
+                        .collect(),
+                    negotiating_connections: self
+                        .initiating_connections
+                        .keys()
+                        .cloned()
+                        .collect(),
+                };
+                let _ = responder.send(status);
+            }
+            NodeApiRequest::Shutdown => {
+                self.shutdown = true;
+                // Shutdown all connection processing tasks
+                for (_, handle) in &self.accepted_connections {
+                    let _ = handle.tx.send(MainToConnMsg::Close).await;
+                }
+                for (_, handle) in &self.initiating_connections {
+                    let _ = handle.tx.send(MainToConnMsg::Close).await;
+                }
+                for (_, handle) in &self.established_connections {
+                    let _ = handle.tx.send(MainToConnMsg::Close).await;
+                }
```

Comment:
Should this be parallelized via `buffer_unordered` or similar?

Reply:
That seems like overkill for local channel sends. I"m not sure it will actually speed things up either, as I don't expect any blocking here. 

A larger question I've thought about is just going ahead and making all these internal channels unbounded, but that makes me worry about memory blowup due to a bug.